app/src/main/java/com/example/perfectweatherallyear/MyFirebaseMessagingService.kt

```diff
@@ -0,0 +1,65 @@
+package com.example.perfectweatherallyear
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.widget.RemoteViews
+import androidx.core.app.NotificationCompat
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+
+const val channelId = "notification_channel"
+const val channelName = "com.example.perfectweatherallyear"
+class MyFirebaseMessagingService : FirebaseMessagingService() {
+
+    override fun onMessageReceived(message: RemoteMessage) {
+        if (message.notification != null) {
+            message.notification!!.title?.let {
+                message.notification!!.body?.let { it1 ->
+                    generateNotification(
+                        it,
+                        it1
+                    )
+                }
+            }
+        }
+    }
+
+    private fun generateNotification(title: String, message: String) {
+        val intent = Intent(this, MainActivity::class.java)
+        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+
+        val pendingIntent = PendingIntent.getActivity(this, 0, intent, PendingIntent.FLAG_ONE_SHOT)
+
+        var builder: NotificationCompat.Builder =
+            NotificationCompat.Builder(applicationContext, channelId)
+                .setSmallIcon(R.drawable.close_black)
+                .setAutoCancel(true)
+                .setOnlyAlertOnce(true)
+                .setContentIntent(pendingIntent)
+
+        builder = builder.setContent(getRemoteView(title, message))
+
+        val notificationManager =
+            getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val notificationChannel =
+                NotificationChannel(channelId, channelName, NotificationManager.IMPORTANCE_HIGH)
+            notificationManager.createNotificationChannel(notificationChannel)
+        }
+        notificationManager.notify(0, builder.build())
+    }
```

Comment:
You have class NotificationHandler for code related to building and posting notifications